### ISSUE TITLE
Accessing unavailable attributes gives a confusing error

### ISSUE THREAD

**Issue by pquentin** [python-trio/trio#1329]
When I access a variable that does not exist in Trio, I get this error message:

```
>>> import trio
>>> trio.stuff
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/quentin/.virtualenvs/connectors-watcher/lib/python3.6/site-packages/trio/_deprecate.py", line 125, in __getattr__
    raise AttributeError(name)
AttributeError: stuff
```

And Sentry displays it like this:

![Capture d’écran 2019-12-03 à 08 59 17](https://user-images.githubusercontent.com/42327/70035673-3f12a980-15ab-11ea-8190-cb7e9073ec5d.png)

Why is it confusing, you might say? The error is raised from `_deprecate.py` and I thought it was only a deprecation warning, but in fact it did not work at all.

I would have expected the standard behavior:

```
>>> import asyncio
>>> asyncio.stuff
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'asyncio' has no attribute 'stuff'
>>>
```

**Comment by njsmith** [comment]
Hmm, that's tricky.

The reason it's being raised from `_deprecate.py` is because `_deprecate.py` checks access to undefined attributes on `trio`, because for deprecated attributes it needs to issue a deprecation warning and then return the real value. But once it's intercepted the attribute lookup to handle deprecated attributes, it *has* to handle it somehow, so if the name *isn't* a deprecated attribute then it has to raise `AttributeError` itself.

I don't know how to make the traceback match what you were expecting :-/ I guess we could do some hack, like drop the first frame from the traceback?

**Comment by pquentin** [comment]
Yes, dropping the first frame would work for me. We already go out of our way to produce beautiful tracebacks in other cases (https://vorpus.org/blog/beautiful-tracebacks-in-trio-v070/), would you consider it to be acceptable in this case? I'm also happy to close the issue as I now understand it.

**Comment by pquentin** [comment]
Closed in #1330 thanks to @dhirschfeld who is now part of the team. :)
